cmd/input-file-available-producer/main.go

```diff
@@ -17,11 +19,12 @@ var kafkaHost = flag.String("kafka", "localhost:9092", "")
 func main() {
 
 	flag.Parse()
+	ctx := context.Background()
 
 	var brokers []string
 	brokers = append(brokers, *kafkaHost)
 
-	producer, _ := kafka.NewProducer(brokers, *topic, int(2000000))
+	producer, _ := kafka.NewProducer(ctx, brokers, *topic, int(2000000), kafka.CreateProducerChannels())
```

Comment:
If you just handle the error here and call os.Exit(1) - this package might be removed at a latter date

Reply:
Also, I will replace the code below `panic(err)` for `os.Exist(1)`